library/src/options.cxx

```diff
@@ -82,6 +120,25 @@ std::vector<std::string> options::getNames() const
   return options_tools::getNames();
 }
 
+//----------------------------------------------------------------------------
+std::vector<std::string> options::getSetNames() const
+{
+  std::vector<std::string> names = this->getNames();
+  std::vector<std::string> setNames;
+  for (const std::string& name : names)
+  {
+    try
+    {
+      options_tools::get(*this, name);
+      setNames.emplace_back(name);
+    }
+    catch (const f3d::options::unset_exception&)
+    {
+    }
+  }
```

Comment:
```suggestion
  {
    if(this->isSet(name))
    {
      setNames.emplace_back(name);
    }
  }
```
logic would be the same but the intent is clearer this way?
I'm sure @Meakk will also have a one-liner with something from `<algorithm>`

Reply:
yeah i implemented isSet last ^^

Reply:
Something like that?
```cpp
  std::copy_if(names.begin(), names.end(), std::back_inserter(setNames),
    [&](const std::string& name){
       return this->isSet(name);
    });
```